### ISSUE TITLE
Incorrect loss calculation in `DMoNPooling`

### ISSUE THREAD

**Issue by xiaohan2012** [pyg-team/pytorch_geometric#10148]
## The bug
In [DMoNPooling](https://pytorch-geometric.readthedocs.io/en/2.6.0/generated/torch_geometric.nn.dense.DMoNPooling.html), calculating the cluster loss (the $$\mathcal{L}_c$$ term in the documentation) requires calculating the graph sizes for a batch of graphs.

The current implementation, however, obtains the graph sizes as a wrongly-shaped tensor, which leads to the wrong cluster loss.

## Context

This pooling operator takes a batch of graphs. For each graph, its cluster loss is defined as 

$$
\frac{\sqrt{C}}{n}
        {\left\|\sum_i\mathbf{C_i}^{\top}\right\|}_F - 1.
$$

where $C$ is the number of clusters, and $\mathbf{C}$ is the node-level cluster matrix.

The corresponding snippet in `DMoNPooling`is:

https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/dense/dmon_pool.py#L147-L151

```python
        cluster_size = torch.einsum('ijk->ik', s)  # B x C
        cluster_loss = torch.norm(input=cluster_size, dim=1)
        cluster_loss = cluster_loss / mask.sum(dim=1) * torch.norm(i_s) - 1
        cluster_loss = cluster_loss.mean()
```

in which:

- variable `s`, a 3D tensor, stores the clustering matrix ($\mathbf{C}$) for the data batch
- variable `mask`, a 2D tensor, stores the presence of nodes in each graph
- and `mask.sum(dim=1)` returns the size of each graph in the batch

## What is expected and what is wrong

`cluster_loss` (before calling `.mean()`) should store the graph-level cluster loss, therefore its shape should be `B` or `Bx1` (because each graph in the batch has one loss value)

The current implementation incorrectly gives a tensor of shape `B x B`.

The reason is that:

- `cluster_loss = torch.norm(input=cluster_size, dim=1)` gives shape `B`
- whereas `mask.sum(dim=1)` gives shape `Bx1`
- `cluster_loss = cluster_loss / mask.sum(dim=1) * torch.norm(i_s) - 1` gives shape `BxB

To fix the bug, we should use `mask.sum(dim=1).squeeze()` to return a shape `B` tensor.

### Versions

```
PyTorch version: 2.4.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.2.1 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.3.9.4)
CMake version: version 3.31.6
Libc version: N/A

Python version: 3.12.8 (main, Dec  3 2024, 18:42:41) [Clang 16.0.0 (clang-1600.0.26.4)] (64-bit runtime)
Python platform: macOS-14.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M2 Pro

Versions of relevant libraries:
[pip3] flake8==7.1.1
[pip3] mypy-extensions==1.0.0
[pip3] numpy==2.1.3
[pip3] pytorch-lightning==2.4.0
[pip3] torch==2.4.0
[pip3] torch_cluster==1.6.3
[pip3] torch-geometric==2.7.0
[pip3] torch_scatter==2.1.2
[pip3] torch_sparse==0.6.18
[pip3] torch_spline_conv==1.2.2
[pip3] torchmetrics==1.6.0
[conda] No relevant packages
```
